chainntnfs/interface.go

```diff
@@ -366,8 +366,8 @@ type BlockEpoch struct {
 	// the main chain.
 	Height int32
 
-	// BlockHeader is the block header of this new height.
-	BlockHeader *wire.BlockHeader
+	// Block is the full block.
+	Block wire.MsgBlock
```

Comment:
I think this would put more pressure on the stack than on the heap so kinda nice. Though it's unclear what's considered as "large", go does recommend [pass values](https://go.dev/wiki/CodeReviewComments#pass-values).